libbeat/processors/registered_domain/registered_domain.go

```diff
@@ -106,6 +106,13 @@ func (p *processor) Run(event *beat.Event) (*beat.Event, error) {
 		return event, errors.Wrapf(err, "failed to write registered domain to target field [%v]", p.TargetField)
 	}
 
+	if p.TargetETLDField != "" {
```

Comment:
The only edge I case I see is that the input value `v` could be an empty string. And that would cause an empty field to be added to the event. Probably the simplest thing to do is check if `tld` is empty before adding the field to the event.

Reply:
So, I actually couldn't create an edge case that changes the behavior. If `v` is an empty string, the first call to `EffectiveTLDPlusOne` winds up with an error and either early returns if `IgnoreFailure` is set, or returns the error directly. There's actually a test case for that in the unit tests (`{true, "", "", "", ""},`).

That said, just for consistency sake I empty string checked the `PublicSuffix` call.